### ISSUE TITLE
Rule Request: redundant_override

### ISSUE THREAD

**Issue by lordzsolt** [realm/SwiftLint#2999]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

When the keyword `required` is added to an initializer, you no longer need to add `override` keyword, because it is already implied.

**1. Why should this rule be added? Share links to existing discussion about what**
   the community thinks about this.

`Required Initializers` section:  https://docs.swift.org/swift-book/LanguageGuide/Initialization.html#ID231 

**2. Provide several examples of what _would_ and _wouldn't_ trigger violations.**

- Triggering examples: 
```swift
class Base {
    required init(name: String) { ... }
}

class Derived: Base {
    ↓override required init(name: String) { ... }
}

class Derived2: Base {
    required ↓override init(name: String) { ... }
}
```

- Non-triggering examples:
```swift
class Base {
    required init(name: String) { ... }
}

class Derived: Base {
    required init(name: String) { ... }
}
```

**3. Should the rule be configurable, if so what parameters should be configurable?**

For now, no.
Later, someone might want to make the inverse of this rule, to require always writing both `override` and `required`. 

**4. Should the rule be opt-in or enabled by default? Why?**

According to the guidelines, it could be enabled by default. However, I haven't found many articles that talk about this, so it can be opt-in also.

**Comment by jpsim** [comment]
Great idea!

**Comment by lordzsolt** [comment]
Apparently there is already a warning produced by default. I could've sworn it it wasn't there.

![Screen Shot 2020-01-22 at 16 02 57](https://user-images.githubusercontent.com/8976891/72905358-b86b8480-3d30-11ea-9250-435d7d4ef897.png)

But with that in mind, I think the issue can be closed.
